tests/server/test_compilerservice.py

```diff
@@ -590,3 +595,129 @@ async def test_compileservice_queue(mocked_compiler_service_block, server, clien
+
+@pytest.fixture
+async def old_compile_report(server_with_frequent_cleanups, environment_for_cleanup):
+    now = datetime.datetime.now()
+    time_of_compile = now - datetime.timedelta(days=30)
+    compile_id = uuid.UUID("c00cc33f-f70f-4800-ad01-ff042f67118f")
+    old_compile = {
+        "id": compile_id,
+        "remote_id": uuid.UUID("c9a10da1-9bf6-4152-8461-98adc02c4cee"),
+        "environment": uuid.UUID(environment_for_cleanup),
+        "requested": time_of_compile,
+        "started": time_of_compile,
+        "completed": time_of_compile,
+        "do_export": True,
+        "force_update": True,
+        "metadata": {"type": "api", "message": "Recompile trigger through API call"},
+        "environment_variables": {},
+        "success": True,
+        "handled": True,
+        "version": 1,
+    }
+    await Compile(**old_compile).insert()
+    new_compile = {**old_compile, "id": uuid.uuid4(), "requested": now, "started": now, "completed": now}
+    await Compile(**new_compile).insert()
+
+    report1 = {
+        "id": uuid.UUID("2baa0175-9169-40c5-a546-64d646f62da6"),
+        "started": time_of_compile,
+        "completed": time_of_compile,
+        "command": "",
+        "name": "Init",
+        "errstream": "",
+        "outstream": "Using extra environment variables during compile \n",
+        "returncode": 0,
+        "compile": compile_id,
+    }
+    report2 = {
+        **report1,
+        "id": uuid.UUID("2a86d2a0-666f-4cca-b0a8-13e0379128d5"),
+        "command": "python -m inmanta.app -vvv export -X",
+        "name": "Recompiling configuration model",
+    }
+    await Report(**report1).insert()
+    await Report(**report2).insert()
+    yield compile_id
+
+
+@pytest.mark.asyncio
+async def test_compileservice_cleanup(
+    server_with_frequent_cleanups, client_for_cleanup, environment_for_cleanup, old_compile_report
+):
+    # There is a new and an older report in the database
+    result = await client_for_cleanup.get_reports(environment_for_cleanup)
+    assert result.code == 200
+    assert len(result.result["reports"]) == 2
+
+    result = await client_for_cleanup.get_report(old_compile_report)
+    assert result.code == 200
+    assert len(result.result["report"]["reports"]) > 0
+
+    compilerslice: CompilerService = server_with_frequent_cleanups.get_slice(SLICE_COMPILER)
+    await compilerslice._cleanup()
+
+    # The old report is deleted after cleanup
+    result = await client_for_cleanup.get_report(old_compile_report)
+    assert result.code == 404
+    reports_after_cleanup = await Report.get_list(compile=old_compile_report)
+    assert len(reports_after_cleanup) == 0
+
+    # The new report is still there
+    result = await client_for_cleanup.get_reports(environment_for_cleanup)
+    assert result.code == 200
+    assert len(result.result["reports"]) == 1
+
+
+@pytest.mark.asyncio
+async def test_compileservice_cleanup_on_trigger(client_for_cleanup, environment_for_cleanup, old_compile_report):
```

Comment:
@wouterdb What is your opinion on this test case? It is rather slow.

Reply:
1. We should mark it as slow  (@pytest.mark.slowtest)

It is indeed slow, but it does test exactly what should be tested, which makes it attractive.

Setting retention to 2 seconds would speed things up, I don't know if it is still stable then (but I would assume it is)